### ISSUE TITLE
False positive `used-before-assignment` when `else` of `try/else` calls `sys.exit()`

### ISSUE THREAD

**Issue by kayhayen** [PyCQA/pylint#7563]
### Bug description

Given this code snippet:

```
    try:
        yamllint.cli.run([document])
    except SystemExit as e:
        lint_result = e.code
    else:
        sys.exit("Error, yamllint didn't raise expected SystemExit exception.")
    if lint_result != 0:
        sys.exit("Error, no lint clean yaml.")

    my_print("OK.", style="blue")
```

Since `sys.exit()` in the else branch is aborting, there is no doubt that `lint_result` must be assigned.

### Configuration

```ini
No configuration
```

### Command used

```shell
pylint m.py
```

### Pylint output

```shell
Warning used-before-assignment for the `lint_result` usage.
```

### Expected behavior

Warning should not be given.

### Pylint version

```shell
pylint 2.15.3
astroid 2.12.10
Python 3.9.2 (default, Feb 28 2021, 17:03:44)
```

### OS / Environment

Debian Buster with PyPI packages

**Comment by Pierre-Sassoulas** [comment]
The issue look similar to #7520.

**Comment by jacobtylerwalls** [comment]
Similar issue as #6790, because the patch (#6791) only handled `return`, not `sys.exit`.

Should be an easy patch for a contributor to extend #6791. Could consider moving `BUILTIN_EXIT_FUNCS` to a common constant file and then use that.
